include/seqan3/alphabet/concept.hpp

```diff
@@ -678,50 +678,107 @@ inline constexpr auto alphabet_size_v = detail::adl::only::alphabet_size_fn<alph
+ *     * move construction shall not be more efficient than copy construction; this implies no dynamic memory (de-)allocation [this is a semantic requirement that cannot be checked]
+ *   3. seqan3::alphabet_size_v needs to be defined for `t`
+ *   4. seqan3::to_rank needs to be defined for objects of type `t`
  *
- * \par Serialisation
+ * See the documentation pages for the respective requirements.
+ * The implications of 2. are that you can always take function arguments of types that model seqan3::Semialphabet
+ * by value.
  *
- * Types that satisfy the concept (and all refinements) can be serialised via SeqAn3
- * serialisation support.
- * The rank value is (de-)serialised, types need not provide any overloads themselves.
+ * It is highly recommended that non-reference types that model this concept, also model:
+ *
+ *   * std::Regular
+ *   * std::is_trivially_copyable
+ *   * seqan3::StandardLayout
  *
- * \par Concepts and doxygen
+ * All alphabets available in SeqAn (with very few exceptions) do so.
  *
- * The requirements for this concept are given as related functions and metafunctions.
- * Types that satisfy this concept are shown as "implementing this interface".
+ * ### Related types
+ *
+ * If a given type `t` models this concept, the following types typically do so, as well:
+ *
+ *   * `t &`
+ *   * `t const`
+ *   * `t const &`
  */
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT Semialphabet = std::StrictTotallyOrdered<t> &&
-                                requires (t v)
+SEQAN3_CONCEPT Semialphabet =
+    std::StrictTotallyOrdered<t> &&
+    std::CopyConstructible<t> &&
+    std::is_nothrow_copy_constructible_v<t> &&
+    requires (t v)
 {
-    // static data members
-    alphabet_size_v<t>;
+    requires seqan3::alphabet_size_v<t> >= 0;
 
-    // conversion to rank
-    requires           noexcept(to_rank(v));
+    { seqan3::to_rank(v) };
+};
+//!\endcond
+
+// ============================================================================
+// WritableSemialphabet
+// ============================================================================
 
-    // assignment from rank
-    requires           noexcept(assign_rank_to(0, v                           ));
-    requires std::Same<decltype(assign_rank_to(0, v                           )), std::remove_reference_t<t> &>;
+/*!\interface seqan3::WritableSemialphabet <>
+ * \brief A refinement of seqan3::Semialphabet that adds assignability.
+ * \extends seqan3::Semialphabet
+ * \ingroup alphabet
+ *
+ * This concept refines seqan3::Semialphabet and adds the requirement to be able to change the value by
+ * assigning a value of the rank representation.
+ *
+ * For a detailed overview of how the different alphabet concepts are related, see
+ * \ref alphabet module.
+ *
+ * ### Concepts and doxygen
+ *
+ * Concepts are shown as "interfaces" in our documentation for technical reasons.
+ * Types that model this concept are shown as "implementing this interface".
```

Comment:
Pheew, I don't think this is a good idea to put this in every concept definition. This needs to be outsourced to some global page with how to read our documentation or so.

Reply:
Hm... ok. I will remove the entries, but I won't add the page as part of this PR.

Reply:
Yes, this is not part of this PR. But relevant for the release though.